### DISCUSSION TITLE
Artisan Scope Modbus Communication Error

### DISCUSSION THREAD

**Discussion by twistedrivermatthew** [artisan-roaster-scope/artisan#1104] [Q&A]
Hello,
I was told to post this question here from a different forum.  I have read many posts of people getting similar errors for different reasons.

I have an Ambex YM-15C coffee roaster. I am trying to get it connected to a laptop through Artisan Scope. I have purchased a DTech USB to RS422 RS485 Serial Port Converter Adapter Cable. I have connected the cable via the CD / CE ports on the Watlow EZ-Zone to the corresponding T/R + T/R - ports on the USB to RS485 adapter.
When it is plugged in the "Active" light is ON.
I have attached pictures of these two things. I cannot find any guidance online from specifically other coffee roasters on how the wiring is supposed to be done. 
When I turn on or start Artisan Scope the "TXD" light blinks.
Also when I start Artisan Scope it says at the top "Connected via MODBUS" for 1-2 seconds before throwing the "Modbus Communication Error" message. The error log does not provide additional information. When I have a serial port monitor on, it detects communication happening.

I'm fairly certain that my issue is a hardware one, I have gone through the software installation multiple times on different laptops. My drivers are up to date and I don't think I need new ones, I have been running Windows 10. 

Let me know if I'm missing something obvious here. 
Thank You!
![IMG_20230215_143027004](https://user-images.githubusercontent.com/125613168/219502138-9ba64062-dd2b-4843-8378-72d19742f270.jpg)
![IMG_20230215_143010982](https://user-images.githubusercontent.com/125613168/219502173-9af1d158-b5ef-41d6-93cc-5317b9db2993.jpg)

**Comment by roasterdave** [comment]
Some common problems that affect Watlow users are as follows.  The baud rate on the Watlow is different from the one set in Artisan.  This is the most likely cause of your problem.  If the Watlow was left in the default Standard Bus communication mode, instead of being set to Modbus RTU, then Artisan will not be able to connect.  Also, as dumb as it sounds, it is worth checking the EZ-Zone model number to be sure the Modbus RTU option is included.

You can access the baud rate and protocol settings from the Watlow front panel buttons and menus.  Refer to the Watlow EZ-Zone PM User's Guide.  

Let us know if you get it running and what you did to resolve it.

**Comment by jsphvrghs** [comment]
I had a similar kind of setup and faced similar kind of errors. What I finally figured was that I needed to locate the exact register that had the values I wanted to display. I used [modpoll ](https://www.modbusdriver.com/modpoll.html#:~:text=modpoll%20is%20a%20command%20line,Linux%20PCs) initially. Then I figured that Artisan also has the "Scan" option under the "Port" options which is more useful as it scans all registers. (However, note that the scan option is broken in the latest release version of Artisan, so you will have to use the previous version if you want to take this route. See #1056 , #1100

Based on the scan, I located the exact MODBUS register where my values were getting written to, and then I configured Artisan MODBUS option to use that register, and it just worked. 

You may also want to check if the FTDI chip on that device is genuine and also get the appropriate driver. Check with [FT_Prog.](https://ftdichip.com/utilities/)

Also as @roasterdave mentioned, check baud rate, word length and the parity settings on your device.
